ftw/labels/browser/labelsjar.py

```diff
@@ -89,7 +89,8 @@ def colors(self):
 
     def _redirect(self):
         response = self.request.RESPONSE
-        return response.redirect(self.context.absolute_url())
+        referer = self.request.get('HTTP_REFERER')
```

Comment:
The referer may or may not be set on the request. I suggest to use the context's absolute_url as fallback by using it as default value:
```python
referer = self.request.get('HTTP_REFERER', self.context.absolute_url())
```